src/soilId/soilIdService.ts

```diff
@@ -17,58 +17,9 @@
 
 import { graphql } from 'terraso-client-shared/graphqlSchema';
 import { SoilIdInputData } from 'terraso-client-shared/graphqlSchema/graphql';
-import { soilDataToIdInput } from 'terraso-client-shared/soilId/soilIdFunctions';
-import { SoilData } from 'terraso-client-shared/soilId/soilIdTypes';
 import * as terrasoApi from 'terraso-client-shared/terrasoApi/api';
 import { Coords } from 'terraso-client-shared/types';
 
-export const fetchSoilMatches = async ({
-  coords,
-  siteId,
-  soilData,
-}: {
-  coords: Coords;
-  siteId?: string;
-  soilData?: SoilData;
-}) => {
-  /*
-   * We call different APIs for different input scenarios.
-   * When soil data is available, we use the data-based soil match API.
-   * When the input is only a location, we use the location-based soil match API.
-   */
-  if (siteId && soilData) {
-    return fetchDataBasedSoilMatches(coords, soilDataToIdInput(soilData)).then(
-      result => {
-        if (result.__typename === 'SoilIdFailure') {
-          return {
-            locationBasedMatches: [],
-            dataBasedMatches: [],
-            failureReason: result.reason,
-          };
-        }
-        return {
-          locationBasedMatches: [],
-          dataBasedMatches: result.matches,
```

Comment:
gratitude: thank you for cleaning this up, i could not brain a better way when i was adding the failure reasons but this is nicer